sdk/attestation/azure-security-attestation/tests/test_attestation_token.py

```diff
@@ -0,0 +1,145 @@
+# coding: utf-8
+
+#-------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#--------------------------------------------------------------------------
+
+import datetime
+from logging import critical
+from typing import Dict
+import unittest
+from cryptography import x509
+from cryptography.hazmat.primitives import hashes, serialization
+from cryptography.hazmat.primitives.asymmetric.dsa import DSAPublicKey
+from devtools_testutils import AzureTestCase, ResourceGroupPreparer, PowerShellPreparer
+import functools
+from cryptography.hazmat.primitives.asymmetric import ec, rsa
+from  cryptography.x509 import BasicConstraints, CertificateBuilder, NameOID, SubjectAlternativeName
+import base64
+import pytest
+from azure.security.attestation import AttestationToken, AttestationSigningKey, TokenValidationOptions
+
+
+class TestAzureAttestationToken(object):
+
+    def test_create_unsecured_token(self):
+        token = AttestationToken(body={"val1":[1,2,3]})
+        assert token.get_body() == {"val1":[1,2,3]}
+
+    def test_create_signing_key(self):
+        key = self._create_rsa_key()
+        cert = self._create_x509_certificate(key, u'test certificate')
+
+        signer = AttestationSigningKey(key, cert)
+        assert signer._certificate.subject==x509.Name([x509.NameAttribute(NameOID.COMMON_NAME, u'test certificate')])
+
+    def test_create_signing_wrong_key(self):
+        """ SigningKey should throw if the key and certificate don't match.
+        """
+        key1 = self._create_rsa_key()
+        cert = self._create_x509_certificate(key1, u'test certificate')
+        key2 = self._create_rsa_key()
+
+        # This should throw an exception, fail if the exception isn't thrown.
+        with pytest.raises(Exception):
```

Comment:
Generally we want to include the exact Exception so we can be sure of the error raised and not some other error is happening.

Reply:
This is the Exception you mentioned above - the exact Exception *is* Exception currently :).